### ISSUE TITLE
RBAC assignments are too broad for OpenShift

### ISSUE THREAD

**Issue by luislezcair** [coder/enterprise-helm#269]
Hi team.

I've been trying to install Coder v1 in OpenShift 4 with a trial license using oc + helm. 
I ran into the following error:

```
Error: UPGRADE FAILED: failed to create resource: roles.rbac.authorization.k8s.io "coder" is forbidden: user "my-user" (groups=["MY AD GROUPS" "system:authenticated:oauth" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:                                                                                                                   
{APIGroups:[""], Resources:["deployments"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                            
{APIGroups:[""], Resources:["networkpolicies"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                        
{APIGroups:[""], Resources:["pods/log"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                               
{APIGroups:["apps"], Resources:["events"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                             
{APIGroups:["apps"], Resources:["networkpolicies"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                    
{APIGroups:["apps"], Resources:["persistentvolumeclaims"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                             
{APIGroups:["apps"], Resources:["pods"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                               
{APIGroups:["apps"], Resources:["pods/exec"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                          
{APIGroups:["apps"], Resources:["pods/log"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                           
{APIGroups:["apps"], Resources:["secrets"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                            
{APIGroups:["apps"], Resources:["serviceaccounts"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                    
{APIGroups:["apps"], Resources:["services"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                           
{APIGroups:["networking.k8s.io"], Resources:["deployments"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                           
{APIGroups:["networking.k8s.io"], Resources:["events"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                
{APIGroups:["networking.k8s.io"], Resources:["persistentvolumeclaims"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                
{APIGroups:["networking.k8s.io"], Resources:["pods"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                                  
{APIGroups:["networking.k8s.io"], Resources:["pods/exec"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                             
{APIGroups:["networking.k8s.io"], Resources:["pods/log"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                              
{APIGroups:["networking.k8s.io"], Resources:["secrets"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                               
{APIGroups:["networking.k8s.io"], Resources:["serviceaccounts"], Verbs:["create" "update" "delete" "deletecollection"]}                                                                                                                       
{APIGroups:["networking.k8s.io"], Resources:["services"], Verbs:["create" "update" "delete" "deletecollection"]
```

My user has namespace-admin privileges so I should be able to create a ServiceAccount with _most_ of the privileges here.
I had to tweak `templates/rbac.yml` like this to make it work:

```yaml
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: coder
  namespace: {{ .Release.Namespace | quote }}
  labels:
    app.kubernetes.io/name: {{ .Chart.Name }}
    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/version: {{ .Chart.AppVersion }}
    app.kubernetes.io/component: {{ include "coder.serviceName" . }}
rules:
  - apiGroups: [""] # "" indicates the core API group
    resources: ["persistentvolumeclaims", "pods", "services", "secrets", "pods/exec", "events", "serviceaccounts"]
    verbs: ["create", "get", "list", "watch", "update", "patch", "delete", "deletecollection"]

  - apiGroups: [""]
    resources: ["pods/log"]
    verbs: ["get"]

  - apiGroups: ["apps"]
    resources: ["deployments"]
    verbs: ["create", "get", "list", "watch", "update", "patch", "delete", "deletecollection"]

  - apiGroups: ["networking.k8s.io"]
    resources: ["networkpolicies"]
    verbs: ["create", "get", "list", "watch", "update", "patch", "delete", "deletecollection"]

  - apiGroups: ["metrics.k8s.io", "storage.k8s.io"]
    resources: ["pods", "storageclasses"]
    verbs: ["get", "list", "watch"]
```

I think the problem is that by default they're too broad and I'm also not sure if all of them actually exist in OpenShift (i.e. create on pods/log).

It would be nice to have this customizable in the values.yml as we do for SecurityContexts so we don't have to maintain a custom Chart with these changes.

**Comment by luislezcair** [comment]
I just found an [open PR](https://github.com/coder/enterprise-helm/pull/252) to solve this problem. I haven't tried but it seems this could be solved in a generic way without needing to customize the values.yml file.

**Comment by ericpaulsen** [comment]
closed via #273. @luislezcair let us know if the updated chart gets things working for you.
